src/Sulu/Bundle/PreviewBundle/Resources/js/containers/Preview/Preview.js

```diff
@@ -98,6 +105,29 @@ class Preview extends React.Component<Props> {
         }
     }
 
+    componentDidUpdate(props: Props) {
+        if (props.formStore.resourceKey !== this.props.formStore.resourceKey) {
+            this.dispose();
+
+            const {
+                formStore: {
+                    resourceKey,
+                },
+                router: {
+                    route: {
+                        options: {
+                            previewResourceKey = null,
+                        },
+                    },
+                },
+            } = this.props;
+
+            this.previewStore.restart(previewResourceKey ? previewResourceKey : resourceKey).then(() => {
```

Comment:
i think i would prefer to destroy the old store and create a new one. i would not expect a `restart` method to have a `resourceKey` argument. also it feels inconsistent to allow for changing the `resourceKey` but not the `id`/`locale`/`webspace`/...

Reply:
we could also do the start and stop here incl. setting the resource key (or as you mentioned) creating a new PreviewStore

Reply:
@nnatter have changed that